internal/controller/etcdcluster_controller_test.go

```diff
@@ -30,55 +28,66 @@ import (
 	operatorv1alpha1 "go.etcd.io/etcd-operator/api/v1alpha1"
 )
 
-var _ = Describe("EtcdCluster Controller", func() {
-	Context("When reconciling a resource", func() {
-		const resourceName = "test-resource"
+func TestEtcdClusterController(t *testing.T) {
+	ctx := t.Context()
 
-		ctx := context.Background()
+	const resourceName = "test-etcd-cluster"
+	typeNamespacedName := types.NamespacedName{
+		Name:      resourceName,
+		Namespace: "etcd-operator-system",
+	}
 
-		typeNamespacedName := types.NamespacedName{
-			Name:      resourceName,
-			Namespace: "default", // TODO(user):Modify as needed
+	etcdcluster := &operatorv1alpha1.EtcdCluster{}
+
+	t.Log("Checking for EtcdCluster resource")
+	err := k8sClient.Get(ctx, typeNamespacedName, etcdcluster)
+	if err != nil && errors.IsNotFound(err) {
+		t.Log("Creating EtcdCluster resource")
+		resource := &operatorv1alpha1.EtcdCluster{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      resourceName,
+				Namespace: "default",
+			},
+		}
+		if createErr := k8sClient.Create(ctx, resource); createErr != nil {
+			t.Fatalf("Failed to create EtcdCluster resource: %v", createErr)
 		}
-		etcdcluster := &operatorv1alpha1.EtcdCluster{}
-
-		BeforeEach(func() {
-			By("creating the custom resource for the Kind EtcdCluster")
-			err := k8sClient.Get(ctx, typeNamespacedName, etcdcluster)
-			if err != nil && errors.IsNotFound(err) {
-				resource := &operatorv1alpha1.EtcdCluster{
-					ObjectMeta: metav1.ObjectMeta{
-						Name:      resourceName,
-						Namespace: "default",
-					},
-					// TODO(user): Specify other spec details if needed.
-				}
-				Expect(k8sClient.Create(ctx, resource)).To(Succeed())
+	} else if err != nil {
+		t.Fatalf("Failed to get EtcdCluster resource: %v", err)
+	}
+
+	// Defer a cleanup function to remove the resource after the test finishes.
+	defer func() {
+		t.Log("Cleaning up the EtcdCluster resource")
+		resource := &operatorv1alpha1.EtcdCluster{}
+		if getErr := k8sClient.Get(ctx, typeNamespacedName, resource); getErr != nil {
+			if !errors.IsNotFound(getErr) {
+				t.Errorf("Failed to get EtcdCluster resource before deletion: %v", getErr)
 			}
```

Comment:
The resource/object should always be present, otherwise there must be something wrong.
```suggestion
			t.Errorf("Failed to get EtcdCluster resource before deletion: %v", getErr)
```